core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -429,6 +427,8 @@ class KafkaApis(val requestChannel: RequestChannel,
       val memoryRecords = partition.records.asInstanceOf[MemoryRecords]
       if (!authorizedTopics.contains(topicIdPartition.topic))
         unauthorizedTopicResponses += topicIdPartition -> new PartitionResponse(Errors.TOPIC_AUTHORIZATION_FAILED)
+      else if (!metadataCache.contains(topicIdPartition.topicPartition))
+        nonExistingTopicResponses += new TopicIdPartition(topicIdPartition.topicId(), topicIdPartition.topicPartition()) -> new PartitionResponse(Errors.UNKNOWN_TOPIC_OR_PARTITION)
```

Comment:
Thanks for the PR. How about doing this instead?
```suggestion
        nonExistingTopicResponses += topicIdPartition -> new PartitionResponse(Errors.UNKNOWN_TOPIC_OR_PARTITION)
```